### DISCUSSION TITLE
Copilot Extension Skillsets Now in Public Preview 11/19

### DISCUSSION THREAD

**Discussion by nhu-do** [community/community#145052] [Copilot Conversations]
![image](https://github.com/user-attachments/assets/2317bbe9-6f7f-46ba-92be-60c1d8d7aa58)

# Announcing Copilot Skillsets Public Preview :rocket:
In September, we launched [Public Beta (Preview) for Copilot Extensions](https://github.blog/changelog/2024-09-17-announcing-the-public-beta-of-github-copilot-extensions-%F0%9F%8E%89/). Today, we're excited to announce the **Public Preview of Skillsets for Copilot Extensions**, a new, lightweight integration method that allows you to more easily build extensions. 🎊 

Our goal as a platform is to empower integrators to extend Copilot's capabilities beyond what is possible today. A large part of reaching this goal is reducing the barrier to entry and providing you the solutions that best fit your needs. Skillsets represents another step in this journey, providing both new and experienced developers with a more lightweight integration method than before. 

## 📈 Faster, simpler, and consistent
- **Develop faster** with skillsets - let Copilot handle the logic, orchestration, and response formatting
- **Integrate more simply** by defining API endpoints
- **Minimal setup**, with no complex server infrastructure required, with the option to use existing APIs
- **Maintain a consistent experience** with Copilot maintaining natural chat interactions automatically

## 💡 What's new?
- Define up to 5 skill endpoints that Copilot can call
- Simple JSON schema configuration
- Quick setup with minimal code required
- Only receive and handle the data that your API needs to fulfill a request

## ❔ How are skillsets different than Extensions?
As an end user, you won't notice a difference. To invoke a skillset, type @ followed by the extension name and your query using natural language. 

Behind the scenes, Copilot handles the all logic, orchestration, and response formatting. Specifically, Copilot:
- Analyzes the query to determine which skill to call
- Structures the API request based on your JSON schema 
- Calls your endpoint to get the data
- Formats and generates the response in chat

Think of this as Level 1 of building extensions. 

## 🤖 Architecture

![image](https://github.com/user-attachments/assets/ed93b20e-7f36-4971-8447-815d66a8f713)

## ✅ Requirements
- Any GitHub Copilot license (Individual, Business, Enterprise)
- For organizational builds: Free, Team, or supported Enterprise Cloud organization types
- Skillsets only apply to extensions build as GitHub Apps, and not VS code chat participants

## 🎉 Getting started
Check out our [documentation](https://docs.github.com/en/copilot/building-copilot-extensions/building-a-copilot-skillset-for-your-copilot-extension/about-copilot-skillsets) to learn how to build your first skillset.

Already built a Copilot Extension? Existing Agent extensions can be converted to skillsets, but one extension cannot be both a skillset and an agent. 

We'd love to hear your feedback as we continue to iterate in Public Preview, and we're excited to see what you build! 

Try skillsets today and provide your feedback to our product team, clients, and builders [here](https://github.com/copilot-extensions/user-feedback/tree/main). :octocat:

**Comment by Paninkiss** [comment]
El lenguaje natural .. deletreo escrito en la meta en base al sonido, identificar cacofonia en el lenguaje oral
Identificar en el sonido oral sin importar el idioma el nivel de vida y el límite hasta donde llega a ser activo su rastro antes de no marchar y continuar con su, Ciclo###

**Comment by Addddsk717** [comment]
![Screenshot_2024-11-24-01-34-02-03_998d3425f9e75a0428f0fabdce419960](https://github.com/user-attachments/assets/9c9ef1eb-234a-43ac-adc5-f2ff5fa25d5f)

**Comment by mohamadrezaasadi308** [comment]
Ok
